File: src/main/java/frc/robot/utils/MovingAverage.java
package frc.robot.utils;

import java.util.List;
import java.util.Queue;

public class MovingAverage {
    private int numValues;

    private double[] values;

    private double total;

    private int index;

    public MovingAverage(int numValues) {
        this.numValues = numValues;
        this.values = new double[numValues];
    }

    public MovingAverage(double[] values) {
        this.values = values;
        numValues = values.length;
    }


    public double addValue(double value) {
        this.index = this.index % this.numValues;
        

        this.total -= this.values[this.index];

        this.total += value;
        
        this.values[this.index] = value;
        this.index++;


        return this.total / this.numValues;


    }

    public double getValue() {
        return this.total / this.numValues;
    }

    
}

File: src/main/java/frc/robot/utils/MovingAverage.kt
package frc.robot.utils

class MovingAverage {
    private var numValues: Int
    private var values: DoubleArray
    private var total = 0.0
    private var index = 0

    constructor(numValues: Int) {
        this.numValues = numValues
        values = DoubleArray(numValues)
    }

    constructor(values: DoubleArray) {
        this.values = values
        numValues = values.size
    }

    fun setAll(value: Double) {
        total = value * numValues
        for (i in 0 until numValues) values[i] = value
    }

    fun addValue(value: Double): Double {
        index = index % numValues
        total -= values[index]
        total += value
        values[index] = value
        index++
        return total / numValues
    }

    val value: Double
        get() = total / numValues
}
